addons/gunbag/functions/fnc_swapGunbag.sqf

```diff
@@ -0,0 +1,34 @@
+#include "script_component.hpp"
+/*
+ * Author: Ir0n1E and mjc4wilton
+ * Swap primary weapon and weapon in gunbag.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Target <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, target] call ace_gunbag_fnc_swapGunbag
+ *
+ * Public: No
+ */
+
+params ["_unit", "_target"];
+
+private _gunbag = backpackContainer _target;
+
+_unit call EFUNC(common,goKneeling);
+
+// play sound
+if (["ace_backpacks"] call EFUNC(common,isModLoaded)) then {
+    [_target, _gunbag] call EFUNC(backpacks,backpackOpened);
+};
+
+[(PROGRESSBAR_TIME * 1.5), _this, {
```

Comment:
why pass _this when you only use the _unit?

Reply:
The toGunbagCallback and fromGunbagCallback methods are both passed the _this array. I believe this is by design in case _target is ever needed to be called from the GunbagCallback methods by someone else. I have kept this one the same for conformity purposes. @commy2 could probably comment more on the reasoning.

Reply:
ok ignore what i said. i forgot how the progressBar function work